presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestMergeAdjacentWindows.java

```diff
@@ -219,25 +217,14 @@ public void testIntermediateProjectNodes()
         return new WindowNode.Specification(ImmutableList.of(planBuilder.symbol(symbolName, BIGINT)), Optional.empty());
     }
 
-    private WindowNode.Function newWindowNodeFunction(String functionName, String... symbols)
+    private WindowNode.Function newWindowNodeFunction(String name, FunctionHandle functionHandle, String... symbols)
     {
         return new WindowNode.Function(
-                new FunctionCall(
-                        QualifiedName.of(functionName),
-                        Arrays.stream(symbols).map(SymbolReference::new).collect(Collectors.toList())),
-                FUNCTION_HANDLE,
-                frame);
-    }
-
-    private WindowNode.Function newWindowNodeFunction(String functionName, Optional<Window> window, String... symbols)
```

Comment:
There's no validation against `window`?

Reply:
`Window` is actually not used by window function. The window info (from `Window` field in `FunctionCall`) will be decomposed into `frame` by `QueryPlanner`: https://github.com/prestodb/presto/blob/b26cd6ff2062d118173e0a7da30faab0d304648d/presto-main/src/main/java/com/facebook/presto/sql/planner/plan/WindowNode.java#L217


So by verifying `frame` is good enough. `window` is actually not used anywhere in the production code. The test should not verify `window` but on `frame`. I don't quite know why originally it verified `window`...